### ISSUE TITLE
Spellcheck & Variable Typos in JavaScript

### ISSUE THREAD

**Issue by ProCode-Software** [microsoft/vscode#181734]
Type: <b>Feature Request</b>

There should be a spellcheck for strings in JavaScript to check for typos and grammar issues (or even probably add ways to rewrite strings powered by AI), and also if a variable or a function is mispelled, show an inline message in the editor for changes (clicking quick fix is too tedious for me), or if the function or variable doesn't exist, add a quick fix suggestion to create a new one.
Thanks!

VS Code version: Code 1.78.0 (252e5463d60e63238250799aef7375787f68b4ee, 2023-05-03T20:09:00.748Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: No

**Comment by IllusionMH** [comment]
There are couple of spellchecker extensions available, if you worry about grammar. They usually provide suggestions. 

In case when you don't have function/variable available l: you should enable `checkJS` in default project settings, so TS would report errors, including ones about using something that us not available (or just typo)

**Comment by ProCode-Software** [comment]
I use vanilla js

**Comment by IllusionMH** [comment]
Yep, VS Code always used TS under the hood for your operation with JS: IntelliSense, signatures help, refactorings, auto-imports, etc.

But IIRC without `checkJS` these identifiers with typos (so basically not declared in that scope) would be just `any` type instead of an error.
https://code.visualstudio.com/docs/nodejs/working-with-javascript#_type-checking-javascript
